lib/streamlit/runtime/credentials.py

```diff
@@ -102,6 +105,52 @@ def email_prompt() -> str:
 }
 
 
+def save_email(email):
+    headers = {
+        "authority": "api.segment.io",
+        "accept": "*/*",
+        "accept-language": "en-US,en;q=0.9",
+        "content-type": "text/plain",
+        "origin": "http://localhost:8501",
+        "referer": "http://localhost:8501/",
+        "sec-fetch-mode": "cors",
+        "sec-fetch-site": "cross-site",
+        "user-agent": "",
+    }
+
+    data = {
+        "timestamp": datetime.now().isoformat(),
+        "integrations": {},
+        "userId": email,
+        "anonymousId": "",
+        "type": "identify",
+        "traits": {
+            "authoremail": email,
+        },
+        "context": {
+            "page": {
+                "path": "/",
+                "referrer": "",
+                "search": "",
+                "title": "",
+                "url": "http://localhost:8501/",
+            },
+            "userAgent": "",
+            "locale": "",
+            "library": {"name": "requests", "version": "1"},
+        },
+        "writeKey": "iCkMy7ymtJ9qYzQRXkQpnAJEq7D4NyMU",
+        "sentAt": datetime.now().isoformat(),
+        "_metadata": {"bundled": ["Segment.io"], "unbundled": [], "bundledIds": []},
+    }
+
+    response = requests.post(
+        "https://api.segment.io/v1/i", headers=headers, data=str(data)
+    )
+
+    response.raise_for_status()
```

Comment:
(oh and we should also have a separate test that asserts that any `requests.post` exception gets caught and logged by `Credentials.save`)

Reply:
Added this test too